src/from_geo.rs

```diff
@@ -260,29 +292,31 @@ mod test {
     #[test]
     fn unclosed_linear_ring() {
         let ls = LineString(coords(vec![(0., 0.), (0., 1.), (1., 2.)]));
-        let geom: GGeom = LineRing(&ls).try_into().unwrap();
+        let geom = GGeom::try_from(&LineRing(&ls)).unwrap();
 
         assert!(geom.is_valid());
         assert!(geom.is_ring().unwrap());
         assert_eq!(geom.get_coord_seq().unwrap().size().unwrap(), 4);
     }
 
     /// a bit tricky
+    ///
     /// a ring should have at least 3 points.
     /// in the case of a closed ring with only element eg:
     ///
     /// let's take a point list: [p1, p2, p1]
     ///
-    /// p1 ----- p2
-    ///  ^-------|
+    ///  ┌───────┐
+    /// p1       p2
+    ///  └───────┘
     ///
     /// we consider it like a 3 points not closed ring (with the 2 last elements being equals...)
     ///
     /// shapely (the python geos wrapper) considers that too
     #[test]
     fn closed_2_points_linear_ring() {
-        let ls = LineString(coords(vec![(0., 0.), (0., 1.), (1., 1.)]));
-        let geom: GGeom = LineRing(&ls).try_into().unwrap();
+        let ls = LineString(coords(vec![(0., 0.), (0., 1.), (0., 0.)]));
```

Comment:
Related to my earlier comment about linear rings, I'm confused by this test and its documentation. It seems to me like we wouldn't want this geometry to be valid.

Reply:
Just to clarify: 

*before* your change, the code made sense to me. I can see why we'd consider it valid after implicitly closing it with a 4th point.

But to me it seems like *after* your change, where p1 == p3, we'd want to consider the geometry invalid.

Reply:
I agree with you, I made the change to make the code consistent with the comments. And the resulting linestring (0,0), (0,1), (0,0) should be invalid..... or not?? I'm not sure. I'm no python guy, but maybe a quick test using shapely would quickly tell us whether this is valid or not

Reply:
Shapely does indeed consider such a linear ring as valid, by adding a fourth point. So your change is more consistent with the original intention.
This test was implemented originally by https://github.com/georust/geos/pull/9/ where we struggled to catch all such corner cases gracefully. At that time, it seemed safer to just imitate the behavior of Shapely.

Reply:
Ok, so I guess the test is just documenting this quirky behavior.

A clearer comment for the test might be:

> A closed ring should have at least 4 points.
> We implicitly close anything which is unclosed *or* which has 3 points, by appending a copy of the first point.
>
> One quirk of this is that we'll "close" an already closed line string if it has 3 points.
> e.g. take a point list: [p1, p2, p1]
> 
>  p1 ----- p2
>   ^-------|
>
> Mirroring Shapely (the python geos wrapper), we consider it like a 3 points unclosed ring - and close it,
> with the 2 last elements being equals, giving us the ring:  `[p1, p2, p1, p1]`

... but from an outside perspective, that behavior seems surprising and wrong. Anyway, it doesn't seem like new behavior, so I wouldn't recommend blocking the PR on this.